data_diff/database.py

```diff
@@ -212,23 +247,48 @@ def query(self, sql_ast: SqlOrStr, res_type: type):
     def enable_interactive(self):
         self._interactive = True
 
-    def _parse_type(self, type_repr: str, datetime_precision: int = None, numeric_precision: int = None) -> ColType:
+    def _convert_db_precision_to_digits(self, p: int) -> int:
+        """Convert from binary precision, used by floats, to decimal precision."""
+        # See: https://en.wikipedia.org/wiki/Single-precision_floating-point_format
+        return math.floor(math.log(2**p, 10))
+
+    def _parse_type(
+        self, type_repr: str, datetime_precision: int = None, numeric_precision: int = None, numeric_scale: int = None
+    ) -> ColType:
         """ """
 
         cls = self.DATETIME_TYPES.get(type_repr)
         if cls:
             return cls(
-                precision=datetime_precision if datetime_precision is not None else DEFAULT_PRECISION,
+                precision=datetime_precision if datetime_precision is not None else DEFAULT_DATETIME_PRECISION,
                 rounds=self.ROUNDS_ON_PREC_LOSS,
             )
 
+        cls = self.NUMERIC_TYPES.get(type_repr)
+        if cls:
+            if issubclass(cls, Integer):
+                # Some DBs have a constant numeric_scale, so they don't report it.
+                # We fill in the constant, so we need to ignore it for integers.
+                return cls(precision=0)
+
+            elif issubclass(cls, Decimal):
+                return cls(precision=numeric_scale)
+
+            assert issubclass(cls, Float)
+            # assert numeric_scale is None
+            return cls(
+                precision=self._convert_db_precision_to_digits(
```

Comment:
For Postgres, `numeric_precision` in `INFORMATION_SCHEMA.columns` is 24, not 23 as I'd normally expect for a 4-byte float. Do you know why?

This causes a potential problem... For Postgres at least, the precision this function gives, and the documentation, aren't congruent for 4-byte, but is for 8-byte:

```python
math.floor(math.log(2**24, 10))
# => 7

>>> math.floor(math.log(2**53, 10))
15
```

![CleanShot 2022-06-21 at 09 49 21@2x](https://user-images.githubusercontent.com/97400/174815700-b2004b3f-5923-418c-83c0-d8b6587dace4.png)

https://www.postgresql.org/docs/current/datatype-numeric.html

Reply:
No idea why..  Double still gives 53..